packages/expo-updates/src/Updates.ts

```diff
@@ -130,6 +130,10 @@ const manualUpdatesInstructions = isUsingExpoDevelopmentClient
  * executed after the `Updates.reloadAsync` method call resolves, since that depends on the OS and
  * the state of the native module and main threads.
  *
+ * In some production use cases, and in the Detox environment, reloadAsync() can
+ * fail or hang if called directly. For these use cases, this API can be called within a setTimeout(),
+ * to allow other JS and native code to complete before the app reloads.
```

Comment:
This guidance is too ambiguous and makes people feel they can't rely on the API. At minimum, we should describe this as a known issue we're working to resolve.

If a setTimeout call fixes this, that likely means there's something that Updates.reloadAsync() depends on that we're not waiting for. I think we need to figure out what that something is, and change the internal implementation of Updates.reloadAsync() to conceptually look like:
```
await preconditionWeDontYetKnow;
...current logic to reload
```